### PR TITLE
feat: support to fetch `consumer_name` or `consumer_id` via `ctx.var`

### PR DESCRIPTION
**Pull request by Firstsawyou**
### What this PR does / why we need it:

 fix: #2297 
### Pre-submission checklist:

* [x] Did you explain what problem does this PR solve? Or what new features have been added?
* [x] Have you added corresponding test cases?
* [ ] Have you modified the corresponding document?
* [x] Is this PR backward compatible?

### CODE REVIEW

**Review by liuhengloveyou — COMMENTED** [review]
good

File: apisix/core/ctx.lua [hunk 1]
```diff
@@ -96,6 +96,9 @@ do
             elseif key == "service_id" then
                 val = ngx.ctx.api_ctx and ngx.ctx.api_ctx.service_id
 
+            elseif key == "consumer_name" then
+                val = ngx.ctx.api_ctx and ngx.ctx.api_ctx.consumer_id
```
Comment (moonming) [inline apisix/core/ctx.lua:100]: It always confuses me, `name` or `id`? we should choose a good name

**Review by moonming — CHANGES_REQUESTED** [review]

File: apisix/core/ctx.lua (hunk 1 above)
Comment (Firstsawyou) [reply apisix/core/ctx.lua:100]: `consumer_id` and `consumer_name` are the same string type value. After processing the "consumer_id" in the core code "ctx.lua" file here, all users will see the "consumer_name".

**Review by moonming — APPROVED** [review]

File: apisix/core/ctx.lua [hunk 2]
```diff
@@ -96,6 +96,9 @@ do
             elseif key == "service_id" then
                 val = ngx.ctx.api_ctx and ngx.ctx.api_ctx.service_id
 
+            elseif key == "consumer_name" then
```
Comment (membphis) [inline apisix/core/ctx.lua:99]: The same value can also be obtained when using `consumer_id`.

Comment (Firstsawyou) [reply apisix/core/ctx.lua:99]: ok, I will add it later.

**Review by membphis — APPROVED** [review]

**Pull request merged**